packages/amplication-git-utils/src/types.ts

```diff
@@ -19,6 +19,31 @@ export enum EnumGitProvider {
   Github = "Github",
 }
 
+export interface Options {
+  provider?: EnumGitProvider;
+  gitOrganization?: RemoteGitOrganization;
+  pullRequestMode?: EnumPullRequestMode;
+  repositoryUrl?: string;
+  repositoryName?: string;
+  isPrivateRepository?: boolean;
+  installationId?: string;
+  organization?: string;
+  owner?: string;
+  files?: Required<Changes["files"]>;
+  commit?: Commit;
+  pullRequestTitle?: string;
+  pullRequestBody?: string;
+  baseBranch?: string;
+  author?: string;
+  filePath?: string;
+  gitResourceMeta?: GitResourceMeta;
+  pullRequestModule?: PullRequestModule[];
+  amplicationWorkspaceId?: string;
+  limit?: number;
+  page?: number;
+  head?: string;
+}
```

Comment:
How about keeping options for the most generic parameters?
I mean in methods like CreatePullRequest or Commit, `files` might make sense, but not in other like createRepo.
It would help developer to know the mandatory params for functions like the above.

Reply:
But then, when I call the provider factory, I would have to pass not only the provider but also the other required params

Reply:
Sorry I was probably not 100% clear in my statement. 
What I meant is to have something that is more clear as methods params like `commits(options, { files, message, author, repo})` but it's ok to start with this.

Reply:
fixed